From dda2388ae88463e408e5aea25a21693e7e7c71f8 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Tue, 9 Oct 2018 15:33:04 +0800
Subject: [PATCH 4805/5242] MLK-19819-1 drm/imx: lcdif: bypass atomic check
 when CRTC will be disabled

commit  39cca380f90e7a1e82694432a673c6725f009719 from
https://source.codeaurora.org/external/imx/linux-imx.git

On 4.14.y kernel branch, the DRM framework has been modified that
when no CONNECTOR attach to CRTC, the fb creation wil be deferred
until some CONNECTOR has been detected via hotplug. And the system
suspend workflow is also affected accordingly, if the CRTC atomic
check fails, the display-subsystem suspend also will be caused to
fail. So bypass the 'bus_format' check when CRTC is going to be
disabled.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/lcdif/lcdif-crtc.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/imx/lcdif/lcdif-crtc.c b/drivers/gpu/drm/imx/lcdif/lcdif-crtc.c
index 457483e..7d83a04 100644
--- a/drivers/gpu/drm/imx/lcdif/lcdif-crtc.c
+++ b/drivers/gpu/drm/imx/lcdif/lcdif-crtc.c
@@ -94,6 +94,12 @@ static int lcdif_crtc_atomic_check(struct drm_crtc *crtc,
 	struct lcdif_crtc *lcdif_crtc = to_lcdif_crtc(crtc);
 	struct imx_crtc_state *imx_crtc_state = to_imx_crtc_state(state);
 
+	/* Don't check 'bus_format' when CRTC is
+	 * going to be disabled.
+	 */
+	if (!state->enable)
+		return 0;
+
 	/* return directly when no devices attached
 	 * to LCDIF to avoid below error messsage to
 	 * make noises in this case.
-- 
1.7.9.5

